### PR TITLE
Feature: Rename JavaScript Foundation Lessons to be more descriptive

### PR DESCRIPTION
**Pull request by nikitarevenco**
## Because

The foundations lesson titled 'Fundamentals Part X' are not descriptive of what they actually teach, and should be renamed.

## This PR

Rename fundamentals parts 1-5 to:

- Variables and operators
- Data types and conditionals
- Function basics
- Arrays and loops
- Object basics 

## Issue

Closes #[27278](https://github.com/TheOdinProject/curriculum/issues/27278) in the curriculum repo.

## Additional Information

Related to PR #[27342](https://github.com/TheOdinProject/curriculum/pull/27342) in the curriculum repo.

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
    - `Feature` - adds new or amends existing user-facing behavior
    - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
    - `Fix` - bug fixes
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] I have verified all tests and linters pass after making these changes.
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If applicable, this PR includes new or updated automated tests

### CODE REVIEW

**Review by JoshDevHub — CHANGES_REQUESTED** [review]
This is a cool improvement! Nice to have actual names for these lessons.

You'll have to make a few more changes for this to function properly:
1. In `db/fixtures/lessons/foundation_lessons.rb`, you've changed the key, the title name, and the description. You also need to change the `github_path`. This is used to pull the lesson content from the curriculum github repo. In your PR to the curriculum repo, you've changed the name of these files, so a github path like `/foundations/javascript_basics/fundamentals-3.md` isn't going to exist.
2. You need to update [this section](https://github.com/TheOdinProject/theodinproject/blob/main/db/fixtures/paths/foundations/seed.rb#L137-L154) of the `db/fixtures/paths/foundations/seed.rb` file. These fetch calls will use the key names defined in the lessons fixture, so a fetch on `'Fundamentals Part 1'` is going to fail. You need to update the relevant lessons to use the new key names.

Let me know if you have any questions about any of this. I suspect Ruby and Rails are unfamiliar territory for you.

**Comment by nikitarevenco** [comment]
> This is a cool improvement! Nice to have actual names for these lessons.
> 
> You'll have to make a few more changes for this to function properly:
> 
> 1. In `db/fixtures/lessons/foundation_lessons.rb`, you've changed the key, the title name, and the description. You also need to change the `github_path`. This is used to pull the lesson content from the curriculum github repo. In your PR to the curriculum repo, you've changed the name of these files, so a github path like `/foundations/javascript_basics/fundamentals-3.md` isn't going to exist.
> 2. You need to update [this section](https://github.com/TheOdinProject/theodinproject/blob/main/db/fixtures/paths/foundations/seed.rb#L137-L154) of the `db/fixtures/paths/foundations/seed.rb` file. These fetch calls will use the key names defined in the lessons fixture, so a fetch on `'Fundamentals Part 1'` is going to fail. You need to update the relevant lessons to use the new key names.
> 
> Let me know if you have any questions about any of this. I suspect Ruby and Rails are unfamiliar territory for you.

Got it. I've made the changes to the paths and also the fetch to call the correct name. 

And yeah, despite the syntax being different than javascript I can mostly read the code which is awesome.

File: db/fixtures/lessons/foundation_lessons.rb [hunk 1]
```diff
@@ -81,18 +81,18 @@ def foundation_lessons
       identifier_uuid: '8a13a21d-453e-4492-8240-58a2c8d9596f',
       installation_lesson: true,
     },
-    'Fundamentals Part 1' => {
-      title: 'Fundamentals Part 1',
-      description: "You'll get a chance to start picking up the programming fundamentals you need to make your webpages dynamic",
+    'Variables and operators' => {
+      title: 'Variables and operators',
+      description: "In this lesson we learn about variables and operators.",
```
Comment (JoshDevHub) [inline db/fixtures/lessons/foundation_lessons.rb:86]:
```suggestion
      description: 'In this lesson we learn about variables and operators.',
```
Need this string in single quotes to make the linter happy.

**Review by JoshDevHub — CHANGES_REQUESTED** [review]
Just one tiny fix, then we should be good to merge.

**Comment by nikitarevenco** [comment]
> Just one tiny fix, then we should be good to merge.

Done

**Review by JoshDevHub — APPROVED** [review]
Thank you for contributing!

**Pull request merged**